### DISCUSSION TITLE
difference between generic and generic_esp32 bin

### DISCUSSION THREAD

**Discussion by markusruebe** [tbnobody/OpenDTU#1090] [Q&A] [answered]
dear all, I am a little bit confused.
there are now generic_esp32 bin beside the generic one I used uptown now.
I run open dtu on an esp32 - can I still use the centric bin file? what are the esp32 bins for?
hope you can help me to understand.

thanks!

**Comment by tbnobody** [comment, marked as answer]
Please see https://github.com/tbnobody/OpenDTU/wiki/Binary-Installation#overview-of-pre-compiled-binaries

**Reply by markusruebe** [reply]
thanks!!
danke für die super schnelle Antwort! echt klasse!

**Reply by NintendoTito** [reply]
Danke auch für den Link, allerdings hilft mir die dort beschriebene Erklärung auch nicht wirklich weiter, da ich nicht verstehe, wie sich die verschiedenen ESP32 genau unterscheiden.

Ich möchte einen ESP32 Wroom32D flashen. Welche Datei benötige ich da?

**Reply by final-option** [reply]
Ich schließe mich mit der Frage an, da ich ja auch Probleme mit meiner DTU habe. 

Meiner Meinung nach ist der verlinkte Eintrag nicht schlüssig:
- im ersten Absatz steht, es gibt aktuell nur die opendtu-generic.bin Firmware
- der Tabelle darunter steht diese aber gelistet als "depricated"
- danach im Absatz dann, man soll das all-in-one bin file flashen, welches aber ein * im Namen hat
- in den Screenshots zu Espressif sind dann plötzlich 4 Dateien zu sehen
- In der Rubrik "Update DTU" im Wiki wird dann wieder das einfache generic.bin file erwähnt

Dazu kommt, das die Handlungsanweisung zu Espressif irgendwie nicht passt: 
- man soll erst ERASE klicken und dann bei Punkten in der Konsole den BOOT Button drücken. Bei mir ist in der Konsole alles schon durchgelaufen bevor ich überhaupt BOOT drücken kann. Also das wartet nix auf meinen Tastendruck. Genauso beim flashen, selbes Spiel.

Das ist alles irgendwie verwirrend für Otto Normalverbraucher: welches file flashe ich denn nun (und wie finde ich das richtige heraus, das zu meiner Hardware gehört? Mit welchem File update ich? Wie finde ich heraus, ob ich vielleicht aus Versehen das falsche file geflasht habe?

Gruß

**Reply by tbnobody** [reply]
> Meiner Meinung nach ist der verlinkte Eintrag nicht schlüssig:
> Dazu kommt, das die Handlungsanweisung zu Espressif irgendwie nicht passt:

Danke für die Hinweise. Habe ich entsprechend angepasst. 

> wie finde ich das richtige heraus, das zu meiner Hardware gehört?

Dazu musst du wissen welche MCU in deiner Hardware verbaut ist. Das kann dir leider niemand abnehmen. (im Falle eines ESP-Wroom32 ist es z.B. ein einfacher ESP32)

> Mit welchem File update ich?

Dies ist nun (hoffentlich) auf den entsprechenden Wiki Seiten beschrieben.

> Wie finde ich heraus, ob ich vielleicht aus Versehen das falsche file geflasht habe?

Das Binary für eine falsche MCU sollte garnicht booten.

**Reply by final-option** [reply]
in der DTU steht unter Info: Chip Model: ESP32-D0WD-V3

**Reply by tbnobody** [reply]
Ich bin mir jetzt nicht ganz sicher was ich mit dieser Information anfangen soll. 
Wenn die DTU das anzeigt, bedeutet das, dass du vmtl. die Firmware erfolgreich geflasht hast.

**Reply by NintendoTito** [reply]
Ok, klingt grundsätzlich gut, mein ESP32 zeigt in Opendtu auch ESP32-D0WD-V3 an. Ich flashte den vor ein paar Monaten mit der bis dahin "normalen" generic.bin. Das läuft auch soweit.

Verstehe ich richtig: Wenn ich jetzt ein Update der Firmware machen möchte, dann nehme ich weiterhin die generic.bin? Und wenn ich die Platine bzw eine baugleiche flashen möchte, dann nehme ich die bin-Datei mit dem ESP32 im Dateinamen?

**Reply by tbnobody** [reply]
> Wenn ich jetzt ein Update der Firmware machen möchte, dann nehme ich weiterhin die generic.bin?

Damit machst du auf jeden Fall erstmal nichts falsch (`opendtu-generic.bin`). Du könntest auch die Datei `opendtu-generic_esp32.bin` verwenden. Wenn du allerdings noch kein [Device Profile](https://github.com/tbnobody/OpenDTU/wiki/Device-Profiles) hochgeladen hast wird das NRF/CMT Modul nicht erkannt werden. Diese neuen `generic` Firmwares haben erstmal keine Pins zugewiesen. Dies wird alles im nachhinein über die json Dateien konfiguriert.

> Und wenn ich die Platine bzw eine baugleiche flashen möchte, dann nehme ich die bin-Datei mit dem ESP32 im Dateinamen?

Das wäre der sauberste Weg. z.B. `opendtu-generic_esp32.factory.bin` via USB flashen, ein entsprechendes Device Profile erstellen oder ein [fertiges](https://github.com/tbnobody/OpenDTU/tree/master/docs/DeviceProfiles) verwenden und hochladen und anschließend unter Einstellungen --> Hardware auswählen.

**Reply by LoeffelXXL** [reply]
Hallo Zusammen,
beim Verifizieren, welche Firmware (opendtu-generic_esp32) nun die richtige für mein Chip-Modell (ESP32-D0WD-V3) ist, hat mir folgende Seite geholfen:
https://www.espressif.com/en/products/socs
Hier kann man auf den Zusammenhang von Chip-Modell, Modul & ESP32 Serie schließen und entsprechend die richtige Firmware wählen.
Hoffe, dem ein oder anderen damit geholfen zu haben.
MfG

**Reply by Ativelos** [reply]
Huhu, ich bin neu in dem ganzen und habe mir bei blinkyparts das OpenDTU CMT Kit geholt um dann zu realisieren, dass ich eigentlich das Kit mit dem NRF Modul brauche. Naiv wie ich bin dachte ich, dass ich die einfach umstecken könnte und dann ist das Problem gelöst. Ist natürlich nicht so einfach. 
Verstehe ich deine Antwort @tbnobody richtig, dass ich opendtu-generic_esp32.bin flashen kann und es dann mit dem entsprechenden device profile der blinkyparts_esp32.json funktionieren müsste?
Vielen Dank im Voraus.

**Reply by Ativelos** [reply]
Also falls irgendjemand so überstürzt gehandelt hat wie ich und dadurch dasselbe Problem hat:
Einfach in den Hardware-Einstellungen das ausgewählte Profil ändern. Fast schon peinlich einfach.

**Comment by NintendoTito** [comment]
Wenn ich jetzt auf einer Platine ESP32-D0WD-V3 die esp32 bin-datei flashe, wie kann ich denn dann über Einstellungen die nodemcu_esp32.json flashen wenn vorher noch keinerlei Pins zugeordnet sind? Dann funktioniert die Weboberfläche doch noch gar nicht, oder?

**Reply by tbnobody** [reply]
> Dann funktioniert die Weboberfläche doch noch gar nicht, oder?

Die Pins an denen z.B. das NRF Modul hängt haben ja nichts mit der Weboberfläche zu tun. Die funktioniert immer.

**Reply by NintendoTito** [reply]
Also das hochladen der nodemcu_esp32.json mache ich dann über die Weboberfläche?

**Comment by final-option** [comment]
Hi.
Ich schaffe es weiterhin nicht (mehr), meine DTU ins Heimnetz zu bekommen.
Wenn ich sie an der Steckdose einstecke, öffnet sich das DTU Wlan und ich lande in der DTU GUI. Hier waren jetzt die letzten Versuche immer die Netzwerkeinstellungen alle vorhanden (SSID, PW, IP Adressen, Gateway etc). Ich sehe in der Liveview und auch der Konsole, dass Daten vom WR kommen.
Nach 3min oder so geht dann der DTU AP aus, und in der Fritzbox taucht die DTU auf - allerdings nur mit 1Mbit Geschwindigkeit. Die Web GUI lässt sich nicht anzeigen im Browser.
Geflasht habe ich zuerst die generic.factory.bin und dann mit der generic.bin ein Update über die DTU GUI. Jeweils die aktuellste Version.

Ich würde jetzt wirklich gerne mal versuchen, den Fehler einzugrenzen, weiß aber nicht wie. Liegt das Verbindungsproblem an der Fritzbox oder der DTU? Soll ich mal an der FB die DTU löschen und neu finden lassen? Keine feste IP durch die FB vergeben lassen? Kann man auf der DTU irgendwas testen hinsichtlich der WLAN Antenne oder so? Soll ich mal was anderes flashen? Wenn ja was?
